src/shared/dlt_multiple_files.c

```diff
@@ -0,0 +1,499 @@
+            }
+        }
+    }
+
+    closedir(dir);
+
+    /* delete file */
+    if (filename_oldest[0]) {
+        if (remove(filename_oldest)) {
+            fprintf(stderr, "Remove file %s failed! error=%s\n", filename_oldest, strerror(errno));
+            return -1; /* ERROR */
+        }
+    } else {
+        fprintf(stderr, "No file to be removed!\n");
+        return -1; /* ERROR */
+    }
+
+    /* return size of deleted file*/
+    return size_oldest;
+}
+
+DltReturnValue multiple_files_buffer_check_size(MultipleFilesRingBuffer *files_buffer)
+{
+    if (files_buffer == NULL) {
+        fprintf(stderr, "multiple files buffer not set\n");
+        return DLT_RETURN_ERROR;
+    }
+
+    struct stat status;
+
+    /* check for existence of buffer files directory */
+    errno = 0;
+    if (stat(files_buffer->directory, &status) == -1) {
+        fprintf(stderr, "Buffer files directory: %s doesn't exist, error=%s\n", files_buffer->directory, strerror(errno));
+        return DLT_RETURN_ERROR;
+    }
+    /* check for accessibility of buffer files directory */
+    else if (access(files_buffer->directory, W_OK) != 0) {
+        fprintf(stderr, "Buffer files directory: %s doesn't have the write access \n", files_buffer->directory);
+        return DLT_RETURN_ERROR;
+    }
+
+    ssize_t total_size = 0;
+    /* check size of complete buffer file */
+    while ((total_size = multiple_files_buffer_get_total_size(files_buffer)) > (files_buffer->maxSize - files_buffer->fileSize)) {
+        /* remove the oldest files as long as new file will not fit in completely into complete multiple files buffer */
+        if (multiple_files_buffer_delete_oldest_file(files_buffer) < 0) return DLT_RETURN_ERROR;
+    }
+
+    return total_size == -1 ? DLT_RETURN_ERROR : DLT_RETURN_OK;
+}
+
+DltReturnValue multiple_files_buffer_open_file_for_append(MultipleFilesRingBuffer *files_buffer) {
+    if (files_buffer == NULL || files_buffer->filenameTimestampBased) return DLT_RETURN_ERROR;
+
+    char newest[NAME_MAX + 1] = {0};
+    char oldest[NAME_MAX + 1] = {0};
+    /* targeting the newest file, ignoring number of files in dir returned */
+
+    if (0 == multiple_files_buffer_storage_dir_info(files_buffer->directory,
+                                                   files_buffer->filenameBase, newest, oldest) ) {
+        // no file for appending found. Create a new one
+        printf("No multiple files for appending found. Create a new one\n");
+        return multiple_files_buffer_create_new_file(files_buffer);
+    }
+
+    char file_path[PATH_MAX + 1];
+    int ret = snprintf(file_path, sizeof(file_path), "%s/%s",
+                         files_buffer->directory, newest);
+
+    if ((ret < 0) || (ret >= NAME_MAX)) {
+        fprintf(stderr, "filename cannot be concatenated\n");
+        return DLT_RETURN_ERROR;
+    }
+
+    /* open DLT output file */
+    errno = 0;
+    files_buffer->ohandle = open(file_path, O_WRONLY | O_APPEND, S_IRUSR | S_IWUSR |
```

Comment:
Hello @danielweber2018 
I would like to remove the superfluous mode bits in PR: https://github.com/COVESA/dlt-daemon/pull/462
Thank you